src/clientAuth/clientAuth.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * @fileoverview Adds authentication information to a request and validates
+ * client info.
+ */
+
+const OICCli = require('../init.js').OICCli;
+const utf8 = require('utf8');
+const base64url = require('base64url');
+
+/**
+ * ClientAuthnMethod
+ * Basic Client Authentication Method class.
+ * @class 
+ * @constructor
+ */
+class ClientAuthnMethod {
+  constructor() {}
+
+  construct(params) {
+    throw new Error('Unsupported Operation Exception');
+  }
+}
+
+function validServiceContext(serviceContext, when) {
```

Comment:
The name of this function and what it does is not really intuitive. At first I thought it will give a valid ServiceContext object, but when I looked at the code it's checking whether `client_secret_expires_at` is set and still before the given time.

How about changing the name into something more meaningful? E.g. `isValidServiceContext` or  `isServiceContextValid`?

It also might make sense to move this function into the `ServiceContext` class or is there a reason to have it here?